### DISCUSSION TITLE
Patch dependencies for Stacked Git

### DISCUSSION THREAD

**Discussion by lthms** [stacked-git/stgit#261] [Ideas]
For the record, [I have taken the time to gather my thoughts about what could Stacked Git patch dependencies looks like](https://soap.coffee/~lthms/posts/StackedGitPatchTheory.html). I have not yet published it, strictly speaking, but I’d be interested to have your thoughts about the workflow I am proposing.

In a nutshell, this proposal can be summarized by

- Stacked Git’s new and edit commands could be extended to let developers declare dependencies between their patches. It would be the commands’ responsibility to enforce the wellfoundness of the dependency graph (e.g., prevent the introduction of cycles in the graph, and maybe diamonds too, at least at first)
- push and pop would automatically take care (pushing or popping) of the selected patch(es) dependencies.

By putting the responsibility for the dependency graph “consistency” on the developer shoulders (rather than on Stacked Git itself), I am under the impression it could limit the amount of work necessary to implement this.

**Comment by tino** [comment]
The link 404's unfortunately.

**Reply by lthms** [reply]
Should be fixed. Thanks!
